src/wallet/wallet2.cpp

```diff
@@ -3550,8 +3613,94 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions(std::vector<crypto
 
 void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>> &outs, const std::list<size_t> &selected_transfers, size_t fake_outputs_count)
 {
-  LOG_PRINT_L2("fake_outputs_count: " << fake_outputs_count);
   outs.clear();
+  
+  if(m_light_wallet && fake_outputs_count > 0) {
+    
+    MDEBUG("LIGHTWALLET - Getting random outs");
+    
+    cryptonote::COMMAND_RPC_LIGHT_WALLET_GET_RANDOM_OUTS::request oreq;
+    cryptonote::COMMAND_RPC_LIGHT_WALLET_GET_RANDOM_OUTS::response ores;
+    
+    size_t light_wallet_requested_outputs_count = (size_t)((fake_outputs_count + 1) * 1.5 + 1);
+    
+    oreq.amounts = {"0"};
+    oreq.count = light_wallet_requested_outputs_count;
+    m_daemon_rpc_mutex.lock();
+    bool r = epee::net_utils::invoke_http_json("/get_random_outs", oreq, ores, m_http_client, std::chrono::seconds(5), "POST");
+    m_daemon_rpc_mutex.unlock();
+    if (!r) {
+      MERROR("Lightwallet: Failed to get random outs");
+    }
```

Comment:
Same comment, there's a function that tests/throws, and this is missing a return/throw.